### PR TITLE
Validate: ContainsEmail and stronger ContainsUrl

### PR DESCRIPTION
**Pull request by bryanpackman**
Refs: adobe-community/issues/issues/21403

### CODE REVIEW

File: util/validate.js [hunk 1]
```diff
@@ -133,10 +133,18 @@ define(function() {
           test: test.bind(/^((?:https?):\/\/)?(?:(?:(?:[\w\.\-\+!$&\'\(\)*\+,;=_]|%[0-9a-f]{2})+:)*(?:[\w\.\-\+%!$&\'\(\)*\+,;=]|%[0-9a-f]{2})+@)?(?:[A-Za-z0-9_\-]+\.)(?:[A-Za-z0-9\-\._])+(?::\d+)?(?:[\/|\?](?:[\w#!:\.\?\+=&@$\'~*,;_\/\(\)\[\]\-]|%[0-9a-f]{2})*)?$/),
           message: 'This field must be a valid URL'
         },
+
+        // Regex from https://stackoverflow.com/questions/10570286/check-if-string-contains-url-anywhere-in-string-using-javascript
         ContainsUrl: {
-          test: test.bind(/(https?:\/\/|www\.)/i),
+          test: test.bind(/([a-zA-Z0-9]+:\/\/)?([a-zA-Z0-9_]+:[a-zA-Z0-9_]+@)?([a-zA-Z0-9.-]+\.[A-Za-z]{2,4})(:[0-9]+)?(\/.*)?/i),
```
[the remark this answers is not in the document]
Comment (bryanpackman) [reply util/validate.js]: should we keep it strictly to http(s) ?

[the remark this answers is not in the document]
Comment (weikinhuang) [reply util/validate.js]: i dunno, that's not for me to choose 😄

[the remark this answers is not in the document]
Comment (celeryclub) [reply util/validate.js]: I can't think of any reason not to block `ftp` too. Can't imagine it's a common problem, but we probably don't want people to link to FTP servers if they aren't allowed to link to HTTP servers.

**Comment by bryanpackman** [comment]
Got confirmation from Zach to block ftp. no reason not to

**Comment by bryanpackman** [comment]
FYI, needed to lock in the version of UUID to avoid testing errors

**Pull request merged**